sbndaq-artdaq/Generators/Common/CAENV1730Readout_generator.cc

```diff
@@ -1552,6 +1555,17 @@ bool sbndaq::CAENV1730Readout::readWindowDataBlocks() {
 	- (fTTT_ns - (long)fMeanPollTimeNS >  500000000) * 1000000000
 	- fTimeOffsetNanoSec;
     }
+    else if(fUseTimeTagShiftForTimeStamp){
+      fTTT = uint32_t{header->triggerTimeTag}; // 
+      fTTT_ns = fTTT*8 - fCAEN.recordLength * fCAEN.postPercent / 100 * 2;
+      
+      // Scheme borrowed from what Antoni developed for CRT.
+      // See https://sbn-docdb.fnal.gov/cgi-bin/private/DisplayMeeting?sessionid=7783
+      fTS = fMeanPollTime - fMeanPollTimeNS + fTTT_ns
+	+ (fTTT_ns - (long)fMeanPollTimeNS < -500000000) * 1000000000
+	- (fTTT_ns - (long)fMeanPollTimeNS >  500000000) * 1000000000
+	- fTimeOffsetNanoSec;
+    }
```

Comment:
Expanding on my comment above, and looking at the fragment generation in `readWindowDataBlocks`:

- The time of the fragment (`fTS`) is built in this way: last **full seconds** from `fMeanPollTime` (computed by subtracting the nanoseconds part from the full timestamp) +  nanoseconds of the TTT - length of post-trigger buffer in ns + possible +-1s correction. 
- _Note_: both the TTT and the buffer length are < 1s, so they are not changing the "seconds" part of the timestamp. On the other hand, the additional correction is +-1s, so it doesn't affect the "nanoseconds" part of the timestamp.
- The original `TTT` remains available in the event header, while `fTS` is saved in the fragment metadata. So both values can be accessed in the offline within the same fragment.
- If you remove the **full seconds** from `fTS` (which would remove the `fMeanPollTime - fMeanPollTimeNS` as well as the +-1s correction) and you also subtract the TTT (read from the header), you can recover the post-trigger length of the buffer.

Given this, it seems that saving the `fCAEN.postPercent` in the metadata is superflous as it can be recovered quickly with information already present in the fragment itself.